### PR TITLE
[ARM plugin] Add external neon kernels

### PR DESCRIPTION
**Pull request by allnes**
- [x] Fix compilation errors on RPi
- [x] Measure performance
- [x] Get legal approval

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
As discussed offline:
- let's move 3rd party code to third party folder
- create readme with links where this code has been taken from
- update appropriate files with legal aspects.  
     - let's also update https://github.com/openvinotoolkit/openvino/blob/master/cmake/packaging/copyrights/arm_cpu

**Comment by allnes** [comment]
@ilya-lavrenov @alvoron please review

**Comment by alvoron** [comment]
I think that the list of 3rd party kernels should be across all frameworks, so the file should be hosted in `external_kernels` directory.

**Comment by allnes** [comment]
> I think that the list of 3rd party kernels should be across all frameworks, so the file should be hosted in `external_kernels` directory.

Done

File: modules/arm_plugin/thirdparty/external_kernels/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME "openvino_arm_cpu_external_kernels")
+
+add_library(${TARGET_NAME} INTERFACE)
+
+target_include_directories(${TARGET_NAME} INTERFACE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/ncnn>
+        $<INSTALL_INTERFACE:ncnn>)
```
Comment (ilya-lavrenov) [inline modules/arm_plugin/thirdparty/external_kernels/CMakeLists.txt:11]: install interface is not needed

File: modules/arm_plugin/thirdparty/external_kernels/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# List of external files that used in [OpenVINO ARM plugin](https://github.com/openvinotoolkit/openvino_contrib/tree/master/modules/arm_plugin)
+
+| **Original link on file**                                                                    | **Original repo** |
+|----------------------------------------------------------------------------------------------|-------------------|
+| [ncnn/neon_math.h](https://github.com/Tencent/ncnn/blob/master/src/layer/arm/neon_mathfun.h) | [NCNN](https://github.com/Tencent/ncnn) |
```
Comment (ilya-lavrenov) [inline modules/arm_plugin/thirdparty/external_kernels/README.md:5]: can we provide de a full link including commit hash ?

**Review by ilya-lavrenov — APPROVED** [review]

File: modules/arm_plugin/thirdparty/external_kernels/CMakeLists.txt (hunk 1 above)
Comment (allnes) [reply modules/arm_plugin/thirdparty/external_kernels/CMakeLists.txt]: @ilya-lavrenov done

File: modules/arm_plugin/thirdparty/external_kernels/README.md (hunk 2 above)
Comment (allnes) [reply modules/arm_plugin/thirdparty/external_kernels/README.md]: @ilya-lavrenov done

**Pull request merged**